MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressHci.c

```diff
@@ -718,29 +890,171 @@ NvmeCreateIoSubmissionQueue (
   return Status;
 }
 
+// MU_CHANGE [BEGIN] - Allocate IO Queue Buffer
+
+/**
+  Initialize the Nvm Express controller Data (IO) Queues
+
+  @param[in] Private                 The pointer to the NVME_CONTROLLER_PRIVATE_DATA data structure.
+  @param[in] Aqa                     The pointer to used to the NVME_AQA data structure.
+
+  @retval EFI_SUCCESS                The NVM Express Controller is initialized successfully.
+  @retval Others                     A device error occurred while initializing the controller.
+
+**/
+EFI_STATUS
+NvmeControllerInitIoQueues (
+  IN NVME_CONTROLLER_PRIVATE_DATA  *Private
+  )
+{
+  UINTN       AsqsPages;
+  UINTN       QueuePairPageCount;
+  UINTN       Index;
+  EFI_STATUS  Status;
+
+  // TODO do we want to store the AsqsPages/QueuePairPageCount in the private data structure? so we don't have to recaclulate it later on?

```

Comment:
I think you can drop the todos.

While it would speed up execution, the speed up would be minimal